File: Library/src/main/java/org/endeavourhealth/library/model/dto/RecentActivityItemDto.java
package org.endeavourhealth.library.model.dto;

import java.util.Date;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class RecentActivityItemDto {

  private String iri;
  private Date dateTime;
  private String action;
}

File: src/interfaces/RecentActivityItemDto.ts
import z from "zod";

// export interface RecentActivityItemDto {
//   iri?: string;
//   dateTime?: Date;
//   action?: string;
// }

export const RecentActivityItemDtoSchema = z.strictObject({
  iri: z.string().optional(),
  dateTime: z.date().optional(),
  action: z.string().optional()
});

export type RecentActivityItemDto = z.output<typeof RecentActivityItemDtoSchema>;
